sipa/utils/__init__.py

```diff
@@ -82,6 +88,38 @@ def support_hotline_available():
         return False
 
 
+@cached(cache=TTLCache(maxsize=1, ttl=300))
+def meetingcal():
+    url = "https://agdsn.de/cloud/remote.php/dav/public-calendars/bgiQmBstmfzRdMeH?export"
+
+    try:
+        response = requests.get(url, timeout=1)
+    except requests.exceptions.RequestException:
+        return []
+        
+    if response.status_code != 200:
+        return []
+    
+    try:
+        calendar = icalendar.Calendar.from_ical(response.text)
+    except ValueError:
+        return []
+
+    events = recurring_ical_events.of(calendar).between(datetime.now(), datetime.now() + relativedelta(months=1))
+
+    next_meetings = []
+    for event in events:
+        next_meetings.append({
+            "title": event["SUMMARY"],
+            "datetime": event["DTSTART"].dt,
+            "location": event["LOCATION"] if "LOCATION" in event else "-",
+            "location_link": markdown.markdown(event["LOCATION"]) if "LOCATION" in event else "-"
+        })
```

Comment:
You might find this a bit nitpicky, but please use a list comprehension instead of a for-loop if you just use it to `append` things once at a time, like this:

```suggestion
    next_meetings = [
        {
            "title": event["SUMMARY"],
            "datetime": event["DTSTART"].dt,
            "location": event["LOCATION"] if "LOCATION" in event else "-",
            "location_link": markdown.markdown(event["LOCATION"])
            if "LOCATION" in event
            else "-",
        }
        for event in events
    ]
```

The reason is not only syntactical, but also practical: In a for loop, you can do much more general things than just element-wise operation, since you can mutate all of the local state in every step, including things you already appended to the list. A list comprehension restricts you to element-wise transformations (essentially, a `map`), and thus protects you from doing anything unwanted.

You can prevent a whole class of stupid bugs this way :-)

Reply:
Resolved in 0cef0ed880dde5b7b832518d73c89992b6d07525